packages/client/src/generation/ts-builders/KeyType.ts

```diff
@@ -0,0 +1,34 @@
+import { AnyTypeBuilder } from './AnyTypeBuilder'
+import { BasicBuilder } from './BasicBuilder'
+import { FunctionType } from './FunctionType'
+import { ObjectType } from './ObjectType'
+import { UnionType } from './UnionType'
+import { Writer } from './Writer'
+
+export class KeyType implements BasicBuilder {
+  constructor(public baseType: AnyTypeBuilder, public key: string) {}
+  write(writer: Writer): void {
+    const needsBraces =
```

Comment:
Should this be something like `hasBraces` or `needsParens` instead?

Also, would it make sense to define this as a method in `BasicBuilder` interface and let each type tell if it needs to be wrapped when indexing, or is not worth it and not going to be useful elsewhere?

Reply:
Thanks, did that in the next commit.